### PR TITLE
[ENH] package with pyproject.tom and lint

### PR DESCRIPTION
**Pull request by Remi-Gau**
- package with pyproject.toml and hatchling: drop versioneer
- add isort / codespell config and to pre-commit
- add isort / codespell / flake8 to pre-commit config
- apply linting via pre-commit
- fix easy errors manually:
  - unused imports
  - f string related errors
  - ...
- and silence others
  - line length
  - ...

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Biggest chunk of line changes comes from dropping all the clutter that versioneer added in the repo and that can now be taken care of by [hatchling](https://hatch.pypa.io/latest/) and [hatchling-VCS](https://github.com/ofek/hatch-vcs?tab=readme-ov-file).

File: reproschema/models/model.py [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+from typing import Dict, List, Optional, Union
+
 from pydantic.version import VERSION as PYDANTIC_VERSION
 
 if int(PYDANTIC_VERSION[0]) >= 2:
-    from pydantic import BaseModel, ConfigDict, Field, field_validator
+    from pydantic import BaseModel, ConfigDict, Field
 else:
-    from pydantic import BaseModel, Field, validator
+    from pydantic import BaseModel, Field
```
Comment (Remi-Gau) [inline reproschema/models/model.py:12]: I do not think this is needed as we are pinned to pydantic >= 2

File: reproschema/redcap2reproschema.py [hunk 2]
```diff
@@ -184,7 +185,7 @@ def parse_field_type_and_value(field):
         elif value_type == "xsd:date" and field_type == "text":
             input_type = "date"
     elif field_type == "yesno":
-        value_type = "xsd:boolen"
+        value_type = "xsd:boolean"
```
Comment (Remi-Gau) [inline reproschema/redcap2reproschema.py:188]: Am I correct this is a typo?
@yibeichan

File: reproschema/models/model.py (hunk 1 above)
Comment (Remi-Gau) [reply reproschema/models/model.py:12]: @djarecka

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+doctests = True
+exclude =
+    **/__init__.py
+    **/tests/*
+    *build/
+    docs/sphinxext/
+    docs/tools/
+    reproschema/_version.py
+max-line-length=79
+extend-ignore = B001, B006, B016, E501, E722, F821
```
Comment (Remi-Gau) [inline .flake8:11]: I can fix those errors but probably in a follow up PR.

File: reproschema/models/model.py (hunk 1 above)
Comment (djarecka) [reply reproschema/models/model.py:12]: it is possible, this part was automatically generated by the linkml

File: reproschema/redcap2reproschema.py (hunk 2 above)
Comment (djarecka) [reply reproschema/redcap2reproschema.py:188]: yes, this should be boolean

File: reproschema/models/model.py (hunk 1 above)
Comment (Remi-Gau) [reply reproschema/models/model.py:12]: OK let's leave for now then.

Comment (djarecka) [reply reproschema/models/model.py:12]: but for now I would leave it like this

File: .flake8 (hunk 3 above)
Comment (yibeichan) [reply .flake8:11]: maybe another PR later

**Pull request merged**